### PR TITLE
[TMVA][SOFIE] Add support for Custom operator in SOFIE

### PR DESCRIPTION
**Pull request by sanjibansg**
This PR adds support for defining, parsing and generating code for custom operator in TMVA SOFIE. SOFIE is being developed following the ONNX standards, but cases may arise when a particular operator is not defined or developed. In those scenarios, an user can still integrate the operator using the custom operator implementation. 

For adding a custom operator in `RModel`, the user has to define the input tensor names, output tensor names, attribute values, output tensor shapes and a header file name. The `Generate()` function for the custom operator must be defined in a separate header file under a namespace, which is called for computing the output tensor values. 

## Checklist:
- [x] `ROperator_Custom` class
- [x] Including usage in generated code
- [x] Tests for ROperator_Custom operator
- [ ] Tutorials explaining the usage

### CODE REVIEW

**Comment by sanjibansg** [comment]
@phsft-bot build just on ROOT-ubuntu2204/default with flags -Dtmva-sofie=On

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2204/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Comment by lmoneta** [comment]
@phsft-bot build just on ROOT-ubuntu2204/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Comment by sanjibansg** [comment]
@phsft-bot build just on ROOT-ubuntu2204/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Comment by sanjibansg** [comment]
@phsft-bot build just on ROOT-ubuntu2204/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

**Comment by sanjibansg** [comment]
@phsft-bot build just on ROOT-ubuntu2204/default, ROOT-ubuntu18.04/default with flags -Dtmva-sofie=On

File: tmva/pymva/test/CMakeLists.txt [hunk 1]
```diff
@@ -128,6 +129,29 @@ endif()
...
+
+   # Testing SOFIE Custom Operator functionality
+   add_executable(emitCustomModel
+                  EmitCustomModel.cxx
+                  ${CMAKE_SOURCE_DIR}/tmva/pymva/src/RModelParser_Keras.cxx
+                  ${CMAKE_SOURCE_DIR}/tmva/sofie/src/RModel.cxx
+                  ${CMAKE_SOURCE_DIR}/tmva/sofie/src/SOFIE_common.cxx
+   )
+   target_link_libraries(emitCustomModel ${PYTHON_LIBRARIES} ${Libraries})
+   target_include_directories(emitCustomModel PRIVATE
```
Comment (lmoneta) [inline tmva/pymva/test/CMakeLists.txt:141]: You need to add these lines (as in the test before) for APPLE :
```
if(APPLE)
   target_link_options(emitCustomModel PRIVATE ${PYTHON_LINK_OPTIONS_Development_Main})
endif()
```

Comment (sanjibansg) [reply tmva/pymva/test/CMakeLists.txt:147]: made the change, thanks!

File: tmva/pymva/test/CMakeLists.txt [hunk 2]
```diff
@@ -136,7 +160,8 @@ endif()
     INCLUDE_DIRS
     ${CMAKE_CURRENT_BINARY_DIR}
    )
-   add_dependencies(TestRModelParserKeras SofieCompileModels_Keras)
+   add_dependencies(TestRModelParserKeras SofieCompileModels_Keras SofieCustomModel)
```
[the remark this answers is not in the document]
Comment (sanjibansg) [reply tmva/pymva/test/CMakeLists.txt:172]: made the change, thanks!

**Review by lmoneta — APPROVED** [review]
LGTM! 
Thank you for the contribution!

**Pull request merged**